src/openrct2/ride/Vehicle.cpp

```diff
@@ -847,10 +847,12 @@ void Vehicle::UpdateMeasurements()
             curRide->max_speed = absVelocity;
         }
 
-        if (curRide->average_speed_test_timeout == 0 && absVelocity > 0x8000)
+        if (curRide->average_speed_test_timeout == 0)
         {
-            curRide->average_speed = AddClamp<int32_t>(curRide->average_speed, absVelocity);
-            stationForTestSegment.SegmentTime++;
+            if (absVelocity > 0x8000)
```

Comment:
Why leave this condition in?

Reply:
You're right, we probably want them both to be under the same condition.